src/viewport/src/Client/ViewportControls.lua

```diff
@@ -25,18 +26,31 @@ function ViewportControls.new(viewport, viewportModel)
 	local self = setmetatable(BaseObject.new(viewport), ViewportControls)
 
 	self._viewportModel = assert(viewportModel, "No rotationYaw")
+	self._enabled = self._maid:Add(ValueObject.new())
 
 	self._maid:GiveTask(self._obj.InputBegan:Connect(function(inputObject)
 		if inputObject.UserInputType == Enum.UserInputType.MouseButton1
 			or inputObject.UserInputType == Enum.UserInputType.MouseButton2
-			or inputObject.UserInputType == Enum.UserInputType.Touch then
+			or inputObject.UserInputType == Enum.UserInputType.Touch
+			and self._enabled.Value then
 			self:_startDrag(inputObject)
 		end
 	end))
 
 	return self
 end
 
+--[=[
+	Sets the enabled state of the controls
+
+	@param enabled boolean
+]=]
+function ViewportControls:SetEnabled(enabled)
+	assert(type(enabled) == "boolean", "Bad enabled")
+
+	self._enabled.Value = enabled
```

Comment:
you could do

```lua
return self._enabled:Mount(enabled)
```

and remove the assert if you wanted to just mount an observable directly here. then in high level call you can do:

```lua
self._controls:SetEnabled(self._isEnabled:Observe())
```

no change necessary. thought you would appreciate that tho.